code/modules/tgui/tgui.dm

```diff
@@ -254,7 +254,7 @@
 		return
 	var/datum/host = src_object.ui_host(user)
 	// If the object or user died (or something else), abort.
-	if(!src_object || !host || !user || !window)
+	if(!QDELETED(src_object) || !QDELETED(host) || !QDELETED(user) || !QDELETED(window))
```

Comment:
How does that work

Reply:
Yeah. I have a feeling the old code was to some degree just catching the nulling that comes from hard deletes. seemed a bit silly